src/pykoala/data_container.py

```diff
@@ -1096,8 +1116,8 @@ def from_fits(cls, path, hdul_extension_map=None, **kwargs):
         - cube: Cube
             An instance of a `pykoala.cubing.Cube`.
         """
-        with fits.open(path) as hdul:
-            return cls(hdul, hdul_extension_map=hdul_extension_map, **kwargs)
+        hdul = fits.open(path)
```

Comment:
I am curious if this change affects the memory storage for a large set of data

Reply:
In principle no, the data is not loaded into memory unless explicitly stated.